updated combatant list so can edit current hp and do math with it

TKDMaster12 opened the pull request:
@Mandalorian007 
updated combatant list so can edit current hp and do math with it
made functions dynamic so they will work with the other edits when added

want to figure out how to get combatants props from encouterDrawers and fill state with that instead of fetching data again. Any ideas?

can't figure out why data wont save to database. It updates information, you can refresh the page go to other pages it still shows new values. But if you stop program and start again it goes back to old data.

not sure why package.json rearranged everything
should have only added "babel-preset-stage-0": "^6.24.1" at the end

Mandalorian007 commented on package.json:
```diff
@@ -31,6 +32,7 @@
     "babel-preset-es2015": "^6.18.0",
-    "babel-preset-react": "^6.16.0"
+    "babel-preset-react": "^6.16.0",
+    "babel-preset-stage-0": "^6.24.1"
```
shouldn't this be version 6.16.0 like the rest of the babel imports?

Mandalorian007 replied:
Also don't care if the order is re-shuffled must be something you run locally, but we can roll with this

Mandalorian007 commented on src/main/js/combatantList.js:
```diff
@@ -46,6 +55,52 @@ class CombatantListUnstyled extends React.Component {
+    this.refreshCombatantsState();
+  }
+
+  refreshCombatantsState() {
```
This shouldn't be duplicated from the encounter drawer

Mandalorian007 commented on src/main/js/combatantList.js:
```diff
@@ -46,6 +55,52 @@ class CombatantListUnstyled extends React.Component {
+      .then(data => this.setState({combatants: data}));
+  }
+
+  updateCombatant(index, data) {
```
This function needs to be pulled up the EncounterDrawer and passed down

Mandalorian007 commented on src/main/js/combatantList.js:
```diff
@@ -63,7 +118,7 @@ class CombatantListUnstyled extends React.Component {
               </TableRow>
             </TableHead>
             <TableBody>
-              {this.props.combatants.map(combatant => {
```
this should stay props and let the top level manage combatant state

TKDMaster12 commented:
@Mandalorian007 
okay. I moved one of the functions over to encounterDrawer
Deleted the copied files and used props to reference them instead

TKDMaster12 commented:
@Mandalorian007 
I removed the index and used combantId . I build all the data for teh updatecompatant function inside the handlecahnge function. 

For some reason it is still not saving the data. if you stop spring-boot and then start up again you see the old data

Mandalorian007 commented on src/main/js/encounterDrawer.js:
```diff
@@ -87,9 +88,36 @@ class EncounterDrawerUnstyled extends React.Component {
+
+    if (!isNaN(value))
+    {
+        let url = 'http://localhost:8080/combatants/' + combatantId;
```
Let's leave the url building in the updateCombatant so all the modification methods hide the url.

Mandalorian007 commented on src/main/js/encounterDrawer.js:
```diff
@@ -87,9 +88,36 @@ class EncounterDrawerUnstyled extends React.Component {
+      .then(() => this.refreshCombatantsState());
+  }
+
+  handleChange(combatantId, dataType, value) {
```
can you move the handle change to the combatantList component and then pass the updateCombatant in?

Mandalorian007 replied:
Trying to isolate the combatantUpdate operations and the individual component flow work.  Considering adding redux to simplify state management longer term

TKDMaster12 replied:
If move the handleChange to the combatantList file, then how do I setState
    this.setState({
       combatants: newState
    });

is there a built in function that i can update parent state from child?

Mandalorian007 commented on src/main/js/encounterDrawer.js:
```diff
@@ -87,9 +88,36 @@ class EncounterDrawerUnstyled extends React.Component {
+
+  handleChange(combatantId, dataType, value) {
+    let data;
+    const newState = this.state.combatants.map(item => {
```
I'm not sure I understand this logic are you calling the url with a patch like:

{
   currentHp: 32
}

TKDMaster12 replied:
yes that is exactly whats happening. 
I am using your CompatnatApI
@PatchMapping("combatants/{combatantId}")
    public Combatant updateCombatant

Mandalorian007 replied:
Okay just built and tested your branch on my computer.

Results:

I hit the API directly to test a modification to the currentHp: Used PostMan
localhost:8080/combatants/326
{
	"currentHp": 1
}

Successfully updated the hp to 1


I launched the UI and played with the text field.  Changing the text field to a new number and refreshing the page (via web browser)  kept the change.

Using math in the field didn't seem to have any effect and it just keep the hit points at whatever the number was.  example: +50 became 50

TKDMaster12 commented:
@Mandalorian007 
add function for delete
Added perception field
Added style to the table

delete button works.  Math not working with out setting state. Seeing some kind of lag in changehandle with out setting the state.

TKDMaster12 commented:
@Mandalorian007 
Finished styling 
added color effect for current hp
I notice lag when updating any fields let me know if that is my computer or the program.

If everything looks good to you, I am ready to merge
